pkg/pillar/dpcreconciler/linuxitems/adapter.go

```diff
@@ -150,6 +150,15 @@ func (c *AdapterConfigurator) Create(ctx context.Context, item depgraph.Item) er
 		c.Log.Error(err)
 		return err
 	}
+	// Make sure the ethernet interface is DOWN before renaming
+	// and changing the MAC address, otherwise we get error
+	// `Device or resource busy`.
+	if err := netlink.LinkSetDown(link); err != nil {
```

Comment:
The code is the same for both cases, what about create a receiver function (like _isAdapterBridgedByNIM()_ ) to perform this action?

Reply:
The function would really just call `netlink.LinkSetDown` and nothing else so it does not save us anything.